homeassistant/components/light/tradfri.py

```diff
@@ -69,44 +98,70 @@ def brightness(self):
         """Return the brightness of the group lights."""
         return self._group.dimmer
 
-    def turn_off(self, **kwargs):
+    @asyncio.coroutine
+    def async_turn_off(self, **kwargs):
         """Instruct the group lights to turn off."""
-        return self._group.set_state(0)
+        result = yield from self._api(self._group.set_state(0))
+        return result
 
-    def turn_on(self, **kwargs):
+    @asyncio.coroutine
+    def async_turn_on(self, **kwargs):
         """Instruct the group lights to turn on, or dim."""
         if ATTR_BRIGHTNESS in kwargs:
-            self._group.set_dimmer(kwargs[ATTR_BRIGHTNESS])
+            yield from self._api(
+                self._group.set_dimmer(kwargs[ATTR_BRIGHTNESS]))
         else:
-            self._group.set_state(1)
+            yield from self._api(self._group.set_state(1))
 
-    def update(self):
-        """Fetch new state data for this group."""
-        self._group.update()
+    def _start_observe(self, err=None):
```

Comment:
Mark things that can be called in the event loop as either `@callback` or a coroutine.